### PR TITLE
feat(core): execution.meta.maskedKeys redaction (Spec 65 §10.3, closes #214)

### LINKED ISSUE #214: Spec 65 §10.3: ExecutionMeta redaction via execution.meta.maskedKeys config
## Scope

Wire `execution.meta.maskedKeys` config to redact sensitive meta keys from execution log records before persistence.

## Why

Phase 2A (PR #213) records meta verbatim on every `ExecutionLogEntry`. Gemini review flagged this as P1 information-leakage risk: developers may accidentally pass PII (`internal_token`, `session_secret`, etc.) via meta, and admin UIs surfacing the execution_log entity would leak it.

## Acceptance

- New config schema entry `execution.meta.maskedKeys: string[]` (defaults to a small built-in list like `['_token', '_secret', 'password']`).
- ActionEngine resolves the config at log time and replaces matching keys with `'[REDACTED]'` (or omits them — pick once).
- Match semantics: exact key + glob? case-sensitive? — pick simplest first (exact, case-insensitive).
- Tests: redacted key produces `[REDACTED]` in log entries (in-memory + Drizzle).

## Refs

- Spec 65 §10.3 Audit logging
- PR #213 (Phase 2A delivery)
- Issue #149

### PR DESCRIPTION
**Pull request by laofahai**
## Summary

Closes #214 — wires \`execution.meta.maskedKeys\` config (Spec 65 §10.3) for execution-log redaction. Resolves the security follow-up gemini flagged on PR #213 Phase 2A.

In-memory \`ctx.meta.get(...)\` continues to return real values during execution; redaction applies **only at log time** before the entry hits ExecutionLogger.

### Surfaces

- **Config** — new \`executionConfig\` schema (\`system:execution\`) with \`meta.maskedKeys: string[]\`. Defaults: \`[\"password\", \"token\", \"secret\", \"api_key\"]\`. Case-insensitive exact-key match. Wired into \`LinchKitConfig\` for typed \`defineConfig\` consumers.
- **Helper** — pure \`redactMetaForLog(meta, maskedKeys)\` in \`types/execution-meta.ts\`. Top-level only (Spec 65 §10.3 talks about \"keys\" not paths). Returns a fresh object without mutating input.
- **ActionEngine** — \`maskedKeys\` resolved once at executor construction, redaction applied inside the \`logExecution()\` helper so every call site (success / failed / blocked / MetaSizeError synthetic frame) is uniformly redacted.

### Cross-model review fix

Both codex and gemini flagged a prototype-pollution issue: a meta payload containing a literal \`__proto__\` key would mutate the output's prototype via \`out[key] = ...\`. Fixed by using \`Object.create(null)\` for the output (commit 054aefe). Two regression tests added covering \`__proto__\` and \`constructor\` keys.

## Test plan

- [x] \`execution-meta.test.ts\` — 12 cases covering redactMetaForLog (identity, exact match, case-insensitive, no mutation, no recursion, prototype pollution, etc.)
- [x] \`execution-logger.test.ts\` — 5 cases covering integration (default list, custom list, handler still sees real value, etc.)
- [x] \`drizzle-execution-logger.test.ts\` — \`***\` round-trip through JSONB column
- [x] All quality gates pass: \`bun run check\` + \`bun run typecheck\` + \`bun test\` (4552 pass / 3 skip / 0 fail)

## Out of scope (v2 follow-ups noted in #214)

- Glob/regex key matching
- Nested-path masking (\`user.password\`)
- Per-entity / per-action override

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Execution logs now automatically redact configured sensitive metadata keys; masked values appear as "***" in persisted logs while handlers see the unredacted meta.
  * System configuration now includes an execution section to customize masked keys and defaults are provided.

* **Tests**
  * Added comprehensive unit and integration tests covering redaction behavior, case-insensitivity, configuration overrides, disabling masking, and edge cases.

### CODE REVIEW

File: packages/core/src/engine/action-engine.ts [hunk 1]
```diff
@@ -280,14 +286,44 @@ export function createActionExecutor(options: ActionExecutorOptions): ActionExec
...
+   *
+   * Resolved at construction (not per-execute) because the registry is
+   * immutable after creation — re-reading on every call would just be lost
+   * work.
+   */
+  const maskedKeys: ReadonlyArray<string> = (() => {
+    if (!configRegistry?.has("system:execution")) {
+      return DEFAULT_EXECUTION_META_MASKED_KEYS;
+    }
+    const cfg = configRegistry.get<{ meta: { maskedKeys: string[] } }>("system:execution");
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/engine/action-engine.ts]: Applied in 824ddb5 — `configRegistry.get` generic now uses `ReadonlyArray<string>` matching the deep-frozen runtime contract.

File: packages/core/src/types/execution-meta.ts [hunk 2]
```diff
@@ -298,6 +298,53 @@ export function extendExecutionMeta(
...
+export function redactMetaForLog(
+  meta: Record<string, unknown> | undefined,
+  maskedKeys: ReadonlyArray<string>,
+): Record<string, unknown> | undefined {
+  if (meta === undefined) return undefined;
+  if (maskedKeys.length === 0) return meta;
+
+  // Build a Set of lowercase masked keys for O(1) per-key lookup.
+  const maskedSet = new Set<string>();
+  for (const k of maskedKeys) maskedSet.add(k.toLowerCase());
```
[the remark this answers is not in the document]
Comment (laofahai) [reply packages/core/src/types/execution-meta.ts]: Applied in 824ddb5 — Set construction simplified to `new Set(maskedKeys.map((k) => k.toLowerCase()))`.
